### PR TITLE
K8SPSMDB-704 - add azure

### PR DESCRIPTION
**Pull request by nmarukovich**

### CODE REVIEW

File: cloud/jenkins/psmdb_operator_aks_version.groovy [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+            steps {
+                installRpms()
+                sh '''
+                    if [ ! -d $HOME/google-cloud-sdk/bin ]; then
+                        rm -rf $HOME/google-cloud-sdk
+                        curl https://sdk.cloud.google.com | bash
+                    fi
+
+                    source $HOME/google-cloud-sdk/path.bash.inc
+                    gcloud components update kubectl
```
Comment (tplavcic) [inline cloud/jenkins/psmdb_operator_aks_version.groovy:164]: Do we need whole google-cloud-sdk, if we only use kubectl maybe we can just download that binary?

File: cloud/jenkins/psmdb_operator_aks_latest.groovy [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+            steps {
+                installRpms()
+                sh '''
+                    if [ ! -d $HOME/google-cloud-sdk/bin ]; then
+                        rm -rf $HOME/google-cloud-sdk
+                        curl https://sdk.cloud.google.com | bash
+                    fi
+
+                    source $HOME/google-cloud-sdk/path.bash.inc
+                    gcloud components update kubectl
```
Comment (tplavcic) [inline cloud/jenkins/psmdb_operator_aks_latest.groovy:164]: Do we need whole google-cloud-sdk, if we only use kubectl maybe we can just download that binary?

**Review by tplavcic — CHANGES_REQUESTED** [review]

File: cloud/jenkins/psmdb-operator-aks-latest.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        default:
+        description: MONGOD image
+    - string:
+        name: IMAGE_BACKUP
+        default:
+        description: Backup image
+    - string:
+        name: IMAGE_PMM
+        default:
+        description: PMM image
```
Comment (tplavcic) [inline cloud/jenkins/psmdb-operator-aks-latest.yml:52]: Please add IMAGE_PMM_SERVER_REPO and IMAGE_PMM_SERVER_TAG options.

File: cloud/jenkins/psmdb-operator-aks-version.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        description: MONGOD image
+    - string:
+        name: IMAGE_BACKUP
+        default:
+        description: Backup image
+    - string:
+        name: IMAGE_PMM
+        default:
+        description: PMM image
+
```
Comment (tplavcic) [inline cloud/jenkins/psmdb-operator-aks-version.yml:51]: Please add IMAGE_PMM_SERVER_REPO and IMAGE_PMM_SERVER_TAG options.

File: cloud/jenkins/psmdb_operator_aks_latest.groovy [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+                            export IMAGE_MONGOD=${IMAGE_MONGOD}
+                        fi
+
+                        if [ -n "${IMAGE_BACKUP}" ]; then
+                            export IMAGE_BACKUP=${IMAGE_BACKUP}
+                        fi
+
+                        if [ -n "${IMAGE_PMM}" ]; then
+                            export IMAGE_PMM=${IMAGE_PMM}
+                        fi
```
Comment (tplavcic) [inline cloud/jenkins/psmdb_operator_aks_latest.groovy:68]: Please add IMAGE_PMM_SERVER_REPO and IMAGE_PMM_SERVER_TAG options.

File: cloud/jenkins/psmdb_operator_aks_latest.groovy [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+            description: 'MONGOD image: perconalab/percona-server-mongodb-operator:main-mongod4.0',
+            name: 'IMAGE_MONGOD')
+        string(
+            defaultValue: '',
+            description: 'Backup image: perconalab/percona-server-mongodb-operator:main-backup',
+            name: 'IMAGE_BACKUP')
+        string(
+            defaultValue: '',
+            description: 'PMM image: perconalab/percona-server-mongodb-operator:main-pmm',
+            name: 'IMAGE_PMM')
```
Comment (tplavcic) [inline cloud/jenkins/psmdb_operator_aks_latest.groovy:148]: Please add IMAGE_PMM_SERVER_REPO and IMAGE_PMM_SERVER_TAG options.

File: cloud/jenkins/psmdb_operator_aks_version.groovy [hunk 7]
```diff
@@ -41,37 +41,39 @@ void runTest(String TEST_NAME) {
...
+                        if [ -n "${IMAGE_BACKUP}" ]; then
+                            export IMAGE_BACKUP=${IMAGE_BACKUP}
+                        fi
 
-                    if [ -n "${IMAGE_PMM}" ]; then
-                        export IMAGE_PMM=${IMAGE_PMM}
-                    fi
+                        if [ -n "${IMAGE_PMM}" ]; then
+                            export IMAGE_PMM=${IMAGE_PMM}
+                        fi
```
Comment (tplavcic) [inline cloud/jenkins/psmdb_operator_aks_version.groovy:68]: Please add IMAGE_PMM_SERVER_REPO and IMAGE_PMM_SERVER_TAG options.

File: cloud/jenkins/psmdb_operator_aks_version.groovy [hunk 8]
```diff
@@ -120,9 +148,7 @@ pipeline {
             name: 'IMAGE_PMM')
```
Comment (tplavcic) [inline cloud/jenkins/psmdb_operator_aks_version.groovy:148]: Please add IMAGE_PMM_SERVER_REPO and IMAGE_PMM_SERVER_TAG options.

File: cloud/jenkins/pxc-operator-aws-openshift.yml [hunk 9]
```diff
@@ -1,15 +0,0 @@
...
-    pipeline-scm:
-      scm:
-        - git:
-            url: https://github.com/Percona-Lab/jenkins-pipelines.git
-            branches:
-            - 'master'
-            wipe-workspace: false
-      lightweight-checkout: true
-      script-path: cloud/jenkins/pxc_operator_aws_openshift.groovy
-
```
Comment (tplavcic) [inline cloud/jenkins/pxc-operator-aws-openshift.yml:15]: Please delete the groovy file and jenkins job also.

File: cloud/jenkins/psmdb-operator-aws-openshift.yml [hunk 10]
```diff
@@ -1,15 +0,0 @@
...
-    pipeline-scm:
-      scm:
-        - git:
-            url: https://github.com/Percona-Lab/jenkins-pipelines.git
-            branches:
-            - 'master'
-            wipe-workspace: false
-      lightweight-checkout: true
-      script-path: cloud/jenkins/psmdb_operator_aws_openshift.groovy
-
```
Comment (tplavcic) [inline cloud/jenkins/psmdb-operator-aws-openshift.yml:15]: Please delete the groovy file and jenkins job also.

**Review by tplavcic — CHANGES_REQUESTED** [review]

File: cloud/jenkins/psmdb_operator_aks_latest.groovy [hunk 11]
```diff
@@ -0,0 +1,314 @@
...
+                        if [ -n "${IMAGE_BACKUP}" ]; then
+                            export IMAGE_BACKUP=${IMAGE_BACKUP}
+                        fi
+
+                        if [ -n "${IMAGE_PMM}" ]; then
+                            export IMAGE_PMM=${IMAGE_PMM}
+                        fi
+                        
+                        if [ -n "${IMAGE_PMM_SERVER_REPO}" ]; then
+                        export IMAGE_PMM_SERVER_REPO=${IMAGE_PMM_SERVER_REPO}
```
Comment (tplavcic) [inline cloud/jenkins/psmdb_operator_aks_latest.groovy:71]: Alignment

File: cloud/jenkins/psmdb_operator_aks_version.groovy [hunk 12]
```diff
@@ -41,37 +41,47 @@ void runTest(String TEST_NAME) {
...
 
-                    if [ -n "${IMAGE_PMM}" ]; then
-                        export IMAGE_PMM=${IMAGE_PMM}
-                    fi
+                        if [ -n "${IMAGE_PMM}" ]; then
+                            export IMAGE_PMM=${IMAGE_PMM}
+                        fi
+                        
+                        if [ -n "${IMAGE_PMM_SERVER_REPO}" ]; then
+                        export IMAGE_PMM_SERVER_REPO=${IMAGE_PMM_SERVER_REPO}
```
Comment (tplavcic) [inline cloud/jenkins/psmdb_operator_aks_version.groovy:71]: Alignment

File: cloud/jenkins/pxc_operator_aws_openshift.groovy [hunk 13]
```diff
@@ -1,379 +0,0 @@
-void IsRunTestsInClusterWide() {
```
Comment (tplavcic) [inline cloud/jenkins/pxc_operator_aws_openshift.groovy:1]: Please delete the groovy file for PSMDB also.

**Review by tplavcic — CHANGES_REQUESTED** [review]

File: cloud/jenkins/psmdb-operator-aks-latest.yml (hunk 3 above)
Comment (tplavcic) [reply cloud/jenkins/psmdb-operator-aks-latest.yml:52]: re-check please, I don't see variables here

File: cloud/jenkins/psmdb-operator-aks-version.yml (hunk 4 above)
Comment (tplavcic) [reply cloud/jenkins/psmdb-operator-aks-version.yml:59]: re-check please, I don't see variables here

File: cloud/jenkins/psmdb-operator-aks-latest.yml [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+    - string:
+        name: IMAGE_PMM_SERVER_REPO
+        default: perconalab/pmm-server
+        description: PMM server image repo
+    - string:
+        name: IMAGE_PMM_SERVER_TAG
+        default: dev-latest
+        description: PMM server image tag
```
Comment (nmarukovich) [inline cloud/jenkins/psmdb-operator-aks-latest.yml:60]: @tplavcic

**Review by tplavcic — APPROVED** [review]

**Pull request merged**